server/protocols/webadmin/webadmin-data/src/test/java/org/apache/james/webadmin/routes/DomainsRoutesTest.java

```diff
@@ -255,6 +262,221 @@ void getDomainShouldReturnNotFoundWhenTheDomainIsAbsent() {
+
+                when()
+                    .get(DOMAIN + "/aliases")
+                .then()
+                    .contentType(ContentType.JSON)
+                    .statusCode(HttpStatus.OK_200)
+                    .body(".", hasSize(0));
+            }
+
+            @Test
+            void deleteShouldReturnNotFoundWhenAliasDomainDoesNotExist() {
+                with().put(DOMAIN);
+
+                when()
+                    .delete(DOMAIN + "/aliases/" + ALIAS_DOMAIN)
+                .then()
+                    .contentType(ContentType.JSON)
+                    .statusCode(HttpStatus.NOT_FOUND_404)
+                    .body("statusCode", is(HttpStatus.NOT_FOUND_404))
+                    .body("type", is("InvalidArgument"))
+                    .body("message", is("The domain list does not contain: " + ALIAS_DOMAIN));
+            }
+
+            @Test
+            void putShouldReturnNotFoundWhenAliasDomainDoesNotExist() {
+                with().put(DOMAIN);
+
+                when()
+                    .put(DOMAIN + "/aliases/" + ALIAS_DOMAIN)
+                .then()
+                    .contentType(ContentType.JSON)
+                    .statusCode(HttpStatus.NOT_FOUND_404)
+                    .body("statusCode", is(HttpStatus.NOT_FOUND_404))
+                    .body("type", is("InvalidArgument"))
+                    .body("message", is("The domain list does not contain: " + ALIAS_DOMAIN));
+            }
+
+            @Test
+            void putShouldNotFailOnExternalDomainAlias() {
+                with().put(DOMAIN);
+
+                when()
+                    .put(EXTERNAL_DOMAIN + "/aliases/" + DOMAIN).prettyPeek()
+                .then()
+                    .contentType(ContentType.JSON)
+                    .statusCode(HttpStatus.NO_CONTENT_204);
+            }
+
+            @Test
+            void deleteShouldNotFailOnExternalDomainDestinationForAnAlias() {
+                with().put(DOMAIN);
+
+                when()
+                    .delete(EXTERNAL_DOMAIN + "/aliases/" + DOMAIN)
+                .then()
+                    .contentType(ContentType.JSON)
+                    .statusCode(HttpStatus.NO_CONTENT_204);
+            }
+
+            @Test
+            void getAliasesShouldListAliasesForExternalDomains() {
+                with().put(DOMAIN);
+
+                with().put(EXTERNAL_DOMAIN + "/aliases/" + DOMAIN);
+
+                when()
+                    .get(EXTERNAL_DOMAIN + "/aliases")
+                .then()
+                    .contentType(ContentType.JSON)
+                    .statusCode(HttpStatus.OK_200)
+                    .body("source", containsInAnyOrder(DOMAIN));
+            }
+
+            @Test
+            void deleteShouldRemoveExternalDomainAlias() {
+                with().put(DOMAIN);
+
+                with().put(EXTERNAL_DOMAIN + "/aliases/" + DOMAIN);
+
+                with().delete(EXTERNAL_DOMAIN + "/aliases/" + DOMAIN);
```

Comment:
Maybe you should create a method to create aliases routes

Reply:
IMO the API is pretty clear - I don't see the benefits method extraction will bring here.